### PR TITLE
[SPARK-14556][SQL] Code clean-ups for package o.a.s.sql.execution.streaming.state

### PR DESCRIPTION
**Pull request by lw-lin**
## What changes were proposed in this pull request?
- `StateStoreConf.**max**DeltasForSnapshot` was renamed to `StateStoreConf.**min**DeltasForSnapshot`
- some state switch checks were added
- improved consistency between method names and string literals
- other comments & typo fix
## How was this patch tested?

N/A

### CODE REVIEW

File: sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/state/StateStoreConf.scala [hunk 1]
```diff
@@ -26,12 +26,11 @@ private[streaming] class StateStoreConf(@transient private val conf: SQLConf) ex
 
   import SQLConf._
 
-  val maxDeltasForSnapshot = conf.getConf(STATE_STORE_MIN_DELTAS_FOR_SNAPSHOT)
```
Comment (lw-lin) [inline sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/state/StateStoreConf.scala]: The current `max...` should be a typo given `STATE_STORE_**MIN**_DELTAS_FOR_SNAPSHOT`

File: sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/state/StateStoreRDD.scala [hunk 2]
```diff
@@ -22,12 +22,12 @@ import scala.reflect.ClassTag
 import org.apache.spark.{Partition, TaskContext}
 import org.apache.spark.rdd.RDD
 import org.apache.spark.sql.types.StructType
-import org.apache.spark.util.{SerializableConfiguration, Utils}
```
Comment (lw-lin) [inline sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/state/StateStoreRDD.scala]: unused imports

File: sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/state/HDFSBackedStateStoreProvider.scala [hunk 3]
```diff
@@ -506,7 +512,6 @@ private[state] class HDFSBackedStateStoreProvider(
       .lastOption
     val deltaBatchFiles = latestSnapshotFileBeforeVersion match {
       case Some(snapshotFile) =>
-        val deltaBatchIds = (snapshotFile.version + 1) to version
```
Comment (lw-lin) [inline sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/state/HDFSBackedStateStoreProvider.scala]: `deltaBatchIds` is not used

File: sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/state/HDFSBackedStateStoreProvider.scala [hunk 4]
```diff
@@ -161,24 +163,27 @@ private[state] class HDFSBackedStateStoreProvider(
       }
     }
 
-    /** Cancel all the updates made on this store. This store will not be usable any more. */
+    /** Abort all the updates made on this store. This store will not be usable any more. */
     override def abort(): Unit = {
+      verify(state == UPDATING || state == ABORTED, "Cannot abort after already committed")
```
Comment (lw-lin) [inline sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/state/HDFSBackedStateStoreProvider.scala]: Let's disallow calling `abort()` on a state store when it has been committed -- deltas or snapshots possibly has been already flushed onto HDFS.

**Comment by lw-lin** [comment]
@srowen @zsxwing would you mind taking a look at this? Thanks! :-)

**Comment by zsxwing** [comment]
LGTM. Merging to master. Thanks, @lw-lin

**Comment by lw-lin** [comment]
@zsxwing thank you for the review & merging ! :-)

**Pull request closed without merging**